com.unity.netcode.gameobjects/Tests/Runtime/Metrics/NetworkVariableMetricsTests.cs

```diff
@@ -25,10 +25,17 @@ public IEnumerator TrackNetworkVariableDeltaSentMetric()
 
             var metricValues = waitForMetricValues.AssertMetricValuesHaveBeenFound();
 
-            var networkVariableDeltaSent = metricValues.First();
-            Assert.AreEqual(nameof(NetworkVariableComponent.MyNetworkVariable), networkVariableDeltaSent.Name);
-            Assert.AreEqual(Server.LocalClientId, networkVariableDeltaSent.Connection.Id);
-            Assert.AreNotEqual(0, networkVariableDeltaSent.BytesCount);
+            bool found = false;
+            foreach (var networkVariableDeltaSent in metricValues)
+            {
+                if (nameof(NetworkVariableComponent.MyNetworkVariable) == networkVariableDeltaSent.Name &&
+                    Client.LocalClientId == networkVariableDeltaSent.Connection.Id &&
+                    0 != networkVariableDeltaSent.BytesCount)
+                {
+                    found = true;
+                }
+            }
+            Assert.IsTrue(found);
```

Comment:
This looks like it should work, seems like there's two pieces of logic changes:
1. Removing requirement that the first metric recorded is the one we're looking for
2. Change requirement from "Sent network variable to server" to "Sent network variable to client"

Both are sane changes, although fixing #1 removes assert granularity which might make it harder to figure out test failures in the future. I think for this test that's not a major risk so I'm OK approving the change.

Nit, could could be simplified as

```
Assert.IsTrue(metricValues.Any(metric =>
    nameof(NetworkVariableComponent.MyNetworkVariable) == networkVariableDeltaSent.Name
    && Client.localClientId == networkVariableDeltaSent.Connection.Id
    && 0 != networkVariableDeltaSent.BytesCount));
```